src/main/app/idam/authorizationMiddleware.ts

```diff
@@ -0,0 +1,65 @@
+import * as express from 'express';
+import * as HttpStatus from 'http-status-codes';
+
+import { JwtExtractor } from 'idam/jwtExtractor';
+import { IdamClient } from 'idam/idamClient';
+import { User } from 'idam/user';
+import { Logger } from '@hmcts/nodejs-logging';
+
+import config from 'config';
+
+const sessionCookieName = config.get<string>('session.cookieName');
+const logger = Logger.getLogger('middleware/authorization');
+
+export function hasTokenExpired (err): boolean {
+  return (err.statusCode === HttpStatus.FORBIDDEN || err.statusCode === HttpStatus.UNAUTHORIZED);
+}
+
+export class AuthorizationMiddleware {
+
+  static requestHandler (
+    requiredRoles: string[],
+    accessDeniedCallback: (req: express.Request, res: express.Response) => void,
+    unprotectedPaths?: string[]): express.RequestHandler {
+
+    return async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+      this.handleUnprotectedPaths(unprotectedPaths, req, next);
```

Comment:
yes necessary. No, not all pages are protected. There are 2 pages in prototype that are unprotected.

Reply:
Are these two pages within 'case/*'?

Reply:
probably no, if they are not case related pages then they will be outside case. We will give more thought when we actually work on that.

Reply:
Would rather see this removed for now. Looks like additional complexity we may not even use.

Reply:
why do you think we may not use? comment above - `yes necessary. No, not all pages are protected. There are 2 pages in prototype that are unprotected.`

Reply:
Because we run IDAM  checks on `app.all('/case/*', this.requestHandler())`. If these two unprotected routes are going be outside of this route anyway then why do we need it?

Reply:
@Jamesnelson117 I am going to leave it in there for now. Please keep in mind that we are in setup phase and will move code around and refactor many times before we have end to end journey working.

Reply:
@Jamesnelson117 @KiranVarma02  there are going to be screening questions which are not going to be protected, I agree with both points - dead code shouldn't be in the repo at any time, but as we know there will be unprotected stuff, can we maybe provide unprotected dummy route, "Hello World! Unprotected" or something?

Reply:
Check `handleUnprotectedPaths` tests in AuthorizationMiddleware, test shows how to add unprotected paths. Is this enough, you think? If i add a dummy route, i have to again write tests for it for coverage. If it's simple if i just remove this code.

Reply:
Stashed it away, removed it from this PR.